ikarus/io/resultevaluators.hh

```diff
@@ -89,13 +126,179 @@ struct PrincipalStress
    * \brief Get the name of the result type (PrincipalStress)
    * \return String representing the name
    */
-  static std::string name() { return "PrincipalStress"; }
+  constexpr static std::string name() { return "PrincipalStress"; }
+
+  /**
+   * \brief Get the number of components in the result
+   * \return Number of components
+   */
+  constexpr static int ncomps() { return dim; }
+};
+
+/**
+ * \brief Struct for calculating Triaxiality stresses
+ * \ingroup resultevaluators
+ * \details The VonMises struct provides a function call operator to calculate von Mises stress.
+ * In 2D, this assumes a plane stress state
+ */
+struct Triaxiality
+{
+  /**
+   * \brief Calculate the result quantity (von Mises stress)
+   * \param resultArray EigenVector containing the stress state in Voigt notation
+   * \param comp component of result (not used here)
+   * \tparam R Type of the matrix
+   * \return Triaxiality stress
+   */
+  template <typename R>
+  double operator()(const R& resultArray, const int comp) const {
+    auto sigeq = VonMises{}(resultArray, 0);
+    auto sigm  = HydrostaticStress{}(resultArray, 0);
+    return sigm / sigeq;
+  }
+  /**
+   * \brief Get the name of the result type
+   * \return String representing the name
+   */
+  constexpr static std::string name() { return "Triaxiality"; }
+
+  /**
+   * \brief Get the number of components in the result  (always 1 for Triaxiality)
+   * \return Number of components
+   */
+  constexpr static int ncomps() { return 1; }
+};
+
+/**
+ * \brief Wrapper to obtain stress results for vanishing materials. It takes a resultevaluator as template argument.
+ * For now this works only for plane strain case.
```

Comment:
Can you also add it for plane stress case? It should be simpler where `sigZ = 0.0` ?